### PR TITLE
Upgrade to knative/pkg 1.9

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

Bump to the latest release of knative/pkg.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Bump knative/pkg to 1.9 and update the Kubernetes minimun version to be v1.24.x
```

### CODE REVIEW

**Comment by vdemeester** [comment]
@tektoncd/core-maintainers this bumps kubernetes_min_version to 0.44, meaning we need to update the version of k8s we provision.
cc @abayer

**Comment by vdemeester** [comment]
Needs https://github.com/tektoncd/plumbing/pull/1332 to be green 😅

**Comment by vdemeester** [comment]
(wrong button again...)

**Comment by vdemeester** [comment]
@tektoncd/core-maintainers it is green 🎉

File: hack/update-deps.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ source $(git rev-parse --show-toplevel)/vendor/github.com/tektoncd/plumbing/scri
 
 cd ${REPO_ROOT_DIR}
 
-VERSION="release-1.8"
```
Comment (lbernick) [inline hack/update-deps.sh:25]: can you add a comment or rename this variable to make it clear that it's the knative version?

**Review by lbernick — COMMENTED** [review]
https://github.com/tektoncd/pipeline#required-kubernetes-version also needs an update, and can you clarify min k8s version in the release notes?

File: hack/update-deps.sh (hunk 1 above)
Comment (vdemeester) [reply hack/update-deps.sh:25]: Done 👼🏼

**Comment by vdemeester** [comment]
@lbernick updated (code, doc and release-note)

**Review by lbernick — APPROVED** [review]

**Pull request merged**